lib/ansible/modules/network/cloudengine/ce_ospf.py

```diff
@@ -0,0 +1,996 @@
+                        msg="Error: network addr is invalid.")
+                if not self.mask:
+                    self.module.fail_json(
+                        msg="Error: network mask must be set.")
+                if not self.mask.isdigit():
+                    self.module.fail_json(
+                        msg="Error: network mask is not digit.")
+                if int(self.mask) < 0 or int(self.mask) > 32:
+                    self.module.fail_json(
+                        msg="Error: network mask is invalid.")
+
+            # area authentication check
+            if self.state == "present" and self.auth_mode:
+                if self.auth_mode == "simple":
+                    if self.auth_text_simple and len(self.auth_text_simple) > 8:
+                        self.module.fail_json(
+                            msg="Error: auth_text_simple is not in the range from 1 to 8.")
+                if self.auth_mode in ["hmac-sha256", "hmac-sha256", "md5"]:
+                    if self.auth_key_id and not self.auth_text_md5:
+                        self.module.fail_json(
+                            msg='Error: auth_key_id and auth_text_md5 should be set at the same time.')
+                    if not self.auth_key_id and self.auth_text_md5:
+                        self.module.fail_json(
+                            msg='Error: auth_key_id and auth_text_md5 should be set at the same time.')
+                    if self.auth_key_id:
+                        if not self.auth_key_id.isdigit():
+                            self.module.fail_json(
+                                msg="Error: auth_key_id is not digit.")
+                        if int(self.auth_key_id) < 1 or int(self.auth_key_id) > 255:
+                            self.module.fail_json(
+                                msg="Error: auth_key_id is not in the range from 1 to 255.")
+                    if self.auth_text_md5 and len(self.auth_text_md5) > 255:
+                        self.module.fail_json(
+                            msg="Error: auth_text_md5 is not in the range from 1 to 255.")
+
+        # process max load balance check
+        if self.state == "present" and self.max_load_balance:
+            if not self.max_load_balance.isdigit():
+                self.module.fail_json(
+                    msg="Error: max_load_balance is not digit.")
+            if int(self.max_load_balance) < 1 or int(self.max_load_balance) > 64:
+                self.module.fail_json(
+                    msg="Error: max_load_balance is not in the range from 1 to 64.")
+
+        # process nexthop weight check
+        if self.nexthop_addr:
+            if not self.is_valid_v4addr(self.nexthop_addr):
+                self.module.fail_json(msg="Error: nexthop_addr is invalid.")
+            if not self.nexthop_weight:
+                self.module.fail_json(
+                    msg='Error: nexthop_addr and nexthop_weight should be set at the same time.')
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```